File: dotNet-sdk/Semantria/Proxy/QueryUpdateProxy.cs
using System;
using System.Collections.Generic;
using System.Text;
using Semantria.Com.Mapping.Configuration;
using Semantria.Com.Mapping.Configuration.Stub;

namespace Semantria.Com.Proxy
{
    public sealed class QueryUpdateProxy : IUpdateProxy<Query>
    {
        #region Constructor

        private QueryUpdateProxy()
        {
        }

        internal static IUpdateProxy<Query> CreateInstance()
        {
            return new QueryUpdateProxy();
        }

        #endregion Constructor

        #region Properties

        private static string REMOVE_ALL_ITEMS_MARK = "*";
        private QueryManagable _managable = new QueryManagable();
        internal QueryManagable Managable
        {
            get { return _managable; }
            set { _managable = value; }
        }

        #endregion Properties

        #region IUpdateProxy<Query> Members 

        object IUpdateProxy<Query>.Stub
        {
            get { return _managable; }
        }

        void IUpdateProxy<Query>.Add(Query obj)
        {
            _managable.Added.Add(obj);
        }

        void IUpdateProxy<Query>.Remove(Query obj)
        {
            _managable.Removed.Add(obj.Name);
        }

        void IUpdateProxy<Query>.Edit(Query obj)
        {
            _managable.Added.Add(obj);
        }

        void IUpdateProxy<Query>.Clone(Query obj) { }

        void IUpdateProxy<Query>.RemoveAll()
        {
            _managable.Added.Clear();
            _managable.Removed.Clear();
            _managable.Removed.Add(REMOVE_ALL_ITEMS_MARK);
        }

        #endregion
    }
}

File: java-sdk/com/semantria/proxies/QueryUpdateProxy.java
package com.semantria.proxies;

import com.semantria.interfaces.IUpdateProxy;
import com.semantria.objects.configuration.Query;

import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlElementWrapper;
import javax.xml.bind.annotation.XmlRootElement;
import java.util.ArrayList;
import java.util.List;

@XmlRootElement(name="queries")
public final class QueryUpdateProxy implements IUpdateProxy<Query>
{
	private List<Query> added;
	private List<String> removed;
	
	public QueryUpdateProxy()
	{
		added = new ArrayList<Query>();
		removed = new ArrayList<String>();
	}
	
	public void add(Query query) 
	{
		added.add(query);
	}
	
	public void remove(Query id)
	{
		removed.add(id.getName());
	}
	
	public void update(Query query)
	{
		added.add(query);
	}

	public void clone(Query item) {}

    public void removeAll() {
        added.clear();
        removed.clear();
        removed.add(REMOVE_ALL_ITEMS_MARK);
    }

	@XmlElementWrapper(name="added")
	@XmlElement(name="query")
	public List<Query> getAdded() { return added; }
	@XmlElementWrapper(name="removed")
	@XmlElement(name="query")
	public List<String> getRemoved() { return removed; }
	
}
